chart/compass/templates/migrator-job.yaml

```diff
@@ -0,0 +1,111 @@
+apiVersion: v1
+data:
+    check-db-connection.sh: |
+        #!/usr/bin/env bash
+        apk add --no-cache postgresql-client
+
+        for var in DB_USER DB_HOST DB_NAME DB_PORT; do
+            if [ -z "${!var}" ] ; then
+                echo "ERROR: $var is not set"
+                discoverUnsetVar=true
+            fi
+        done
+        if [ "${discoverUnsetVar}" = true ] ; then
+            exit 1
+        fi
+
+        for i in {1..30}
+        do
+            pg_isready -U "$DB_USER" -h "$DB_HOST" -p "$DB_PORT" -d "$DB_NAME"
+            if [ $? -eq 0 ]
+            then
+                exit 0
+            fi
+            sleep 1
+        done
+
+        exit 1
+kind: ConfigMap
+metadata:
+    annotations:
+        "helm.sh/hook": post-install,post-upgrade
+        "helm.sh/hook-weight": "-1"
+        "helm.sh/hook-delete-policy": before-hook-creation
+    labels:
+        job: compass-migration
+    name: compass-migration-scripts
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+    name: compass-migration
+    labels:
+        app: {{ .Chart.Name }}
+        release: {{ .Release.Name }}
+    annotations:
+        "helm.sh/hook": post-install,post-upgrade
+        "helm.sh/hook-weight": "0"
+        "helm.sh/hook-delete-policy": before-hook-creation
+spec:
+    template:
+        metadata:
+            labels:
+                app: {{ .Chart.Name }}
+                release: {{ .Release.Name }}
+        spec:
+            shareProcessNamespace: true
+            restartPolicy: OnFailure
```

Comment:
usually in Kyma we use `restartPolicy` set to `Never`, but I would like to know how it will behave if `restartPolicy` is set to `OnFailure`. Does chart installation will fail if this job will keep failing?

Reply:
Eventually, helm will return error:
```
Error: timed out waiting for the condition
```

But I decided to change this value to `Never` because as it's pointed in [this](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/?origin_team=T99LHPS1L#pod-backoff-failure-policy) doc it's easier to debug when failed containers are not terminated.